### DISCUSSION TITLE
proxy run without any dependences

### DISCUSSION THREAD

**Discussion by yanickxia** [linkerd/linkerd2#7471] [General]
We are very interested in proxy, but no doc talk about run proxy without linkerd-controller.
run linkerd2-proxy with yaml/json/...   such as envoy with yaml.

**Comment by adleong** [comment]
Hi @yanickxia.  The Linkerd2-proxy is designed to run as part of the Linkerd2 service mesh.  There is not an easy way to run it in a stand-alone way.

**Reply by yanickxia** [reply]
Yeah, i want to debug proxy, must be run in a kube cluster replace proxy, attend remote debug...
it's too difficulty, any plan to make a mock service to run proxy easy? 😊

**Reply by olix0r** [reply]
The proxy's integration tests implement mocked controllers and run the proxy in a standalone mode. You may be able to run write new integration tests to aid debugging or modify this setup for yourself.

Generally, when debugging the proxy, though, it's far easier (for me) to run in an actual cluster environment, so I run locally with k3d and enter the container with the standard linux tools (nsenter/cgexec/etc).
